Lib/test/_test_multiprocessing.py

```diff
@@ -5760,6 +5760,41 @@ def test_namespace(self):
         self.run_worker(self._test_namespace, o)
 
 
+class TestNamedResource(unittest.TestCase):
+
+    def test_global_named_resource_spawn(self):
+        #
+        # Check that global named resources in main module
+        # will not leak by a subprocess, in spawn context.
+        #
+        import tempfile as tf
+        py = tf.NamedTemporaryFile('w', delete=False)
+        try:
+            py.write('''if 1:
+                import multiprocessing as mp
+
+                ctx = mp.get_context('spawn')
+
+                global_resource = ctx.Semaphore()
+
+                def submain(): pass
+
+                if __name__ == '__main__':
+                    p = ctx.Process(target=submain)
+                    p.start()
+                    p.join()
+            ''')
+            py.close()
+
+            p = subprocess.run([sys.executable, py.name],
+                               stderr=subprocess.PIPE,
+                               text=True)
```

Comment:
We have a helper function `assert_python_ok` that does extra setup required in fringe cases, can you use it?

Reply:
I've changed the code to use `assert_python_ok` instead.